src/Swarm/Game/Value.hs

```diff
@@ -14,6 +15,16 @@ import Swarm.Game.Entity
 import Swarm.Game.Robot
 import Swarm.Language.Value
 
+-- * Patterns
+
+type VRect = Value
+pattern VRect :: Integer -> Integer -> Integer -> Integer -> VRect
+pattern VRect x1 y1 x2 y2 = VPair (VPair (VInt x1) (VInt y1)) (VPair (VInt x2) (VInt y2))
+
+{-# COMPLETE VRect #-}
```

Comment:
I don't think we want this `COMPLETE` pragma.  The meaning of such a pragma would be that pattern-matching on `VRect` is sufficient to cover all possible `Value`s, but that is certainly not true.  And indeed, the `rectCells` function is in fact a partial function, and we don't want to just paper over that with a bogus `COMPLETE` pragma.  We never want partial functions (even in cases where bad inputs "shouldn't ever happen"), since as a matter of policy/quality control we want to try to make sure that the game can not possibly crash.